app/Utils/passwordUtil.php

```diff
@@ -1,40 +1,36 @@
 <?php
 
+declare(strict_types=1);
+
 class FreshRSS_password_Util {
 	// Will also have to be computed client side on mobile devices,
 	// so do not use a too high cost
-	const BCRYPT_COST = 9;
+	public const BCRYPT_COST = 9;
 
 	/**
 	 * Return a hash of a plain password, using BCRYPT
-	 *
-	 * @param string $passwordPlain
-	 * @return string
 	 */
-	public static function hash($passwordPlain) {
+	public static function hash(string $passwordPlain): string {
 		$passwordHash = password_hash(
 			$passwordPlain,
 			PASSWORD_BCRYPT,
 			array('cost' => self::BCRYPT_COST)
 		);
-		$passwordPlain = '';
 
 		// Compatibility with bcrypt.js
 		$passwordHash = preg_replace('/^\$2[xy]\$/', '\$2a\$', $passwordHash);
 
-		return $passwordHash == '' ? '' : $passwordHash;
+		return $passwordHash === '' ? '' : $passwordHash;
```

Comment:
`preg_replace()` may return null. `==` was better

Reply:
Every time we use the non-strict comparison, a little fluffy kitten cries out in heartbreaking distress.... ...joking aside, there are huge differences between the soft comparison and the hard comparison. Also the results are not the same between php7 and PHP8 https://www.php.net/manual/en/types.comparisons.php

it is humanly impossible to know all the subtleties of the non-strict comparison

it is possible to simplify like this:

```suggestion
		if ($passwordHash === '' || $passwordHash === null) {
		    return '';
		}
		return $passwordHash;
```

it is more explicit

;)

Reply:
String comparisons are for strict types, and it is dangerous to replace existing comparisons blindly and without covering all the cases. Most comparisons we have in the current code base have been considered and reviewed.